processing/src/main/java/org/apache/druid/java/util/common/guava/ParallelMergeCombiningSequence.java

```diff
@@ -668,11 +688,16 @@ protected void compute()
         }
       }
       catch (Throwable t) {
-        closeAllCursors(pQueue);
         cancellationGizmo.cancel(t);
-        outputQueue.offer(ResultBatch.TERMINAL);
+        cleanup();
       }
     }
+
+    private void cleanup()
+    {
+      closeAllCursors(pQueue);
+      outputQueue.offer(ResultBatch.TERMINAL);
```

Comment:
oops, i meant to modify this to do a non-blocking offer

clearing the entire queue seems unnecessary since everything has been modified to check for cancellation, but also don't think this cleanup should be blocking, so it should only put the terminal marker in there if it can without blocking (which is what the other action does in its cleanup)